Fix some timing code bit rot, plus minor cleanup.

pkukol opened the pull request:
@dotnet/jit-contrib PTAL

CarolEidt commented on src/jit/compiler.cpp:
```diff
@@ -1190,12 +1190,12 @@ void Compiler::compShutdown()
+        {
+            CompTimeSummaryInfo::s_compTimeSummary.Print(jitTimeLogFile);
+            fclose(jitTimeLogFile);
+        }
```
Since you're here, you should change `NULL` to `nullptr`.

pkukol replied:
Oops, old bad habits die hard - will fix. Thanks!

CarolEidt commented:
LGTM

BruceForstall commented on src/jit/compiler.cpp:
```diff
@@ -7086,8 +7098,14 @@ void CompTimeSummaryInfo::Print(FILE* f)
+
+        // Show slop if it's over a certain percentage of the total
+        double pslop_pct = 100.0*m_total.m_parentPhaseEndSlop*1000.0/countsPerSec/totTime_ms;
+        if (pslop_pct >= 1.0)
```
coding convention: please surround operators by a space

pkukol replied:
Will do - sorry about that.

BruceForstall commented on src/jit/compiler.cpp:
```diff
@@ -7225,8 +7249,8 @@ void JitTimer::PrintCsvHeader()
         fprintf(fp, "\"CPS\"\n");
-        fclose(fp);
     }
+    fclose(fp);
```
Looks like you're closing the read handle, but not the write handle. I assume _waccess() didn't work for some reason?

pkukol replied:
That's what happens when you try to use notepad to apply changes (to avoid the white space issue); glad you caught it, will fix. And yes, _waccess isn't available outside of Windows (well it is, with a slightly different name, but it doesn't appear in PAL).

BruceForstall commented on src/jit/compiler.cpp:
```diff
@@ -7202,7 +7225,8 @@ void JitTimer::PrintCsvHeader()
 
-    if (_waccess(jitTimeLogCsv, 0) == -1)
+    FILE* fp = _wfopen(jitTimeLogCsv, W("r"));
+    if (fp == nullptr)
```
I'm confused: the code already looks like what you've submitted as a PR. You shouldn't submit the same change again. Are you not up-to-date in your source tree?

pkukol replied:
I am now also confused - I've no idea why something would appear twice (if I understand your comment correctly). Maybe I should just squash everything when I commit / push the latest changes, since there seems to be no benefit in preserving the little pieces.

jkotas replied:
> Maybe I should just squash everything when I commit / push the latest changes

Right. It will also make the PR mergeable - it has conflicts in its current state.

pkukol replied:
I didn't realize that this conflicts with parjong's change; I'll retest on Windows and Linux (and OS X in a bit, I don't have that set up yet). This snippet is a wee bit "subtle", if you get my drift, and I think that we need to preserve the existing semantics (e.g. not just append since that might mix entries from multiple runs, as the comments explain).

pkukol commented:
@dotnet-bot test OSX x64 Checked Build and Test

pkukol commented:
So ... @dotnet/jit-contrib now happy with this, or do I need to change anything? I think I've addressed all the comments but I might have missed (or messed up) something so please don't be shy. Thanks!

pgavlin commented:
It looks like this still needs to be rebased against the latest changes. LGTM otherwise.

pkukol commented:
Thanks, Pat - I somehow managed to rebase from the wrong base commit, so nothing actually got pushed to origin; sorry about that, should be fixed now. If / when all the builds go fine, this should be finally ready to merge.

pkukol commented:
@dotnet-bot test Innerloop Linux ARM Emulator Release Cross Build

pkukol commented:
@dotnet-bot help

pkukol commented:
@dotnet-bot test Linux ARM Emulator Cross Release Build

pkukol commented:
@dotnet-bot test Linux ARM Emulator Cross Debug Build

pkukol commented:
@dotnet-bot test Ubuntu x64 Checked Build and Test
@dotnet-bot test OSX x64 Checked Build and Test
@dotnet-bot test Linux ARM Emulator Cross Debug Build

pkukol commented:
@dotnet-bot test Linux ARM Emulator Cross Debug Build

pkukol commented:
@dotnet-bot test FreeBSD x64 Checked Build

pkukol commented:
@dotnet-bot test FreeBSD x64 Checked Build

pkukol commented:
@dotnet-bot test Linux ARM Emulator Cross Debug Build

pkukol commented:
@dotnet-bot test Linux ARM Emulator Cross Release Build

pgavlin commented:
LGTM
